modules/http/src/main/java/org/mule/module/http/internal/listener/HttpListenerRegistry.java

```diff
@@ -213,11 +216,16 @@ private void validateCollision(ListenerRequestMatcher newListenerRequestMatcher)
          */
         public RequestHandler findRequestHandler(final HttpRequest request)
         {
-            final String pathName = normalizePathWithSpacesOrEncodedSpaces(request.getPath());
-            Preconditions.checkArgument(pathName.startsWith(SLASH), "path parameter must start with /");
+            Preconditions.checkArgument( request.getPath().startsWith(SLASH), "path parameter must start with /");
+            final String pathName = getDecodedPathIfDecodable(request.getPath());
```

Comment:
Why not just call `decode` here with a try catch directly? You would avoid having interpretations of what an empty path means and the check for emptiness.